drivers/rtc/rtc_emul.c

```diff
@@ -28,15 +21,9 @@ struct rtc_emul_alarm {
 struct rtc_emul_data {
 	bool datetime_set;
 
-	struct rtc_time datetime;
-
-	struct k_mutex lock;
-
-	struct rtc_emul_work_delayable dwork;
```

Comment:
Usage of the work queue item is preferred to a timer since work, like invoking a callback, shall not be performed from within a timer callback. Work must be delegated to a thread, like the work queue. This is the reason ISR callbacks delegate work to be performed to the work queue.

Keep in mind that this driver is not time critical, it is not a real piece of hardware meant to actually be used as a functional RTC, its an emulator, which provides similar behavior to an RTC, allowing for rapid application development if real hardware is not available.

Reply:
Timer conversion is now removed from change set.